### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by brenthosie**
## Description

- Updates the dependencies
- Adds github actions

## Related Issue

https://jira.corp.adobe.com/browse/DTM-15548

## Motivation and Context

- Resolve security warnings.
- Keep the project up-to-date
- Fix publishing to `npm` using new github actions

## How Has This Been Tested?

- `npm test`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
```
Comment (brenthosie) [inline .github/workflows/node.js.yml:1]: As far as I can tell, we only need to run `test` and don't need to integrate with saucelabs?

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
```
Comment (brenthosie) [inline .github/workflows/npm-publish.yml:1]: Run tests, then publish

File: bin/index.js [hunk 3]
```diff
@@ -11,8 +11,9 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  ****************************************************************************************/
-
-const argv = require('yargs')
+const {hideBin} = require('yargs/helpers');
```
Comment (brenthosie) [inline bin/index.js:14]: These changes are necessary to break the 16.x version boundary for `yargs`

File: package.json [hunk 4]
```diff
@@ -24,22 +24,22 @@
...
   "dependencies": {
-    "@adobe/jwt-auth": "^0.3.0",
+    "@adobe/jwt-auth": "^0.3.3",
     "chalk": "^4.1.0",
-    "delay": "^4.2.0",
-    "inquirer": "^7.3.2",
-    "ora": "^4.0.4",
+    "delay": "^4.4.0",
+    "inquirer": "^7.3.3",
+    "ora": "^5.3.0",
```
Comment (brenthosie) [inline package.json:31]: `5.x` "breaking change" was requiring `node 10`, which we already had in `package.json` of this repo.

File: package.json [hunk 5]
```diff
@@ -24,22 +24,22 @@
...
   },
   "dependencies": {
-    "@adobe/jwt-auth": "^0.3.0",
+    "@adobe/jwt-auth": "^0.3.3",
     "chalk": "^4.1.0",
-    "delay": "^4.2.0",
-    "inquirer": "^7.3.2",
-    "ora": "^4.0.4",
+    "delay": "^4.4.0",
+    "inquirer": "^7.3.3",
```
Comment (brenthosie) [inline package.json:30]: minor updates

File: package.json [hunk 6]
```diff
@@ -24,22 +24,22 @@
     "test:watch": "nodemon --exec jasmine JASMINE_CONFIG_PATH=jasmine.json"
   },
   "dependencies": {
-    "@adobe/jwt-auth": "^0.3.0",
+    "@adobe/jwt-auth": "^0.3.3",
     "chalk": "^4.1.0",
-    "delay": "^4.2.0",
-    "inquirer": "^7.3.2",
-    "ora": "^4.0.4",
+    "delay": "^4.4.0",
```
Comment (brenthosie) [inline package.json:29]: minor updates

File: package.json [hunk 7]
```diff
@@ -24,22 +24,22 @@
     "test:watch": "nodemon --exec jasmine JASMINE_CONFIG_PATH=jasmine.json"
   },
   "dependencies": {
-    "@adobe/jwt-auth": "^0.3.0",
+    "@adobe/jwt-auth": "^0.3.3",
```
Comment (brenthosie) [inline package.json:27]: asked project maintainers to update their dependencies

File: package.json [hunk 8]
```diff
@@ -24,22 +24,22 @@
...
-    "ora": "^4.0.4",
+    "delay": "^4.4.0",
+    "inquirer": "^7.3.3",
+    "ora": "^5.3.0",
     "prettyjson": "^1.2.1",
     "request": "^2.88.0",
     "request-debug": "^0.2.0",
     "request-promise-native": "^1.0.7",
-    "yargs": "^15.4.1",
+    "yargs": "^16.2.0",
```
Comment (brenthosie) [inline package.json:36]: fixed `bin/index.js` to get to `16.x`

File: package.json [hunk 9]
```diff
@@ -24,22 +24,22 @@
...
     "yauzl": "^2.10.0"
   },
   "devDependencies": {
-    "jasmine": "^3.4.0",
-    "jasmine-expect": "^4.0.1",
-    "nodemon": "^2.0.4",
+    "jasmine": "^3.6.4",
```
Comment (brenthosie) [inline package.json:40]: minor updates

File: package.json [hunk 10]
```diff
@@ -24,22 +24,22 @@
...
+    "jasmine-expect": "^5.0.0",
+    "nodemon": "^2.0.7",
```
Comment (brenthosie) [inline package.json:42]: minor updates

File: package.json [hunk 11]
```diff
@@ -24,22 +24,22 @@
...
-    "yargs": "^15.4.1",
+    "yargs": "^16.2.0",
     "yauzl": "^2.10.0"
   },
   "devDependencies": {
-    "jasmine": "^3.4.0",
-    "jasmine-expect": "^4.0.1",
-    "nodemon": "^2.0.4",
+    "jasmine": "^3.6.4",
+    "jasmine-expect": "^5.0.0",
```
Comment (brenthosie) [inline package.json:41]: some matchers (that we aren't using) are [changing](https://github.com/JamieMason/Jasmine-Matchers/compare/4.0.3...5.0.0).

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (Aaronius) [reply .github/workflows/node.js.yml:1]: Yeah, since this is a node script, there's no need to test on browsers.

Comment (Aaronius) [reply .github/workflows/node.js.yml:1]: You might want to check out [this cache action](https://github.com/adobe/alloy/blob/37f94b60e64bad1f2950c9df527b5841287f738b/.github/workflows/test-cron.yaml#L14-L18) to optimize the installation of dependencies. I don't know a lot about it, but it seems nice.

**Review by Aaronius — APPROVED** [review]
Looks good. Thanks for the notes!

**Pull request merged**